builtin/providers/aws/resource_aws_kms_alias.go

```diff
@@ -46,7 +61,16 @@ func resourceAwsKmsAlias() *schema.Resource {
 
 func resourceAwsKmsAliasCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).kmsconn
-	name := d.Get("name").(string)
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else if v, ok := d.GetOk("name_prefix"); ok {
+		name = resource.PrefixedUniqueId(v.(string))
+	} else {
+		name = resource.UniqueId()
+	}
+
```

Comment:
since we're now modifying what we get from the config regarding the name, we need to make sure we use the `d.Id()` in `resourceAwsKmsAliasRead`, because `d.Get("name")` may now return `""`